service/src/main/java/io/camunda/service/VariableServices.java

```diff
@@ -54,19 +54,20 @@ public SearchQueryResult<VariableEntity> search(final VariableQuery query) {
 
   public SearchQueryResult<VariableEntity> search(
       final Function<Builder, ObjectBuilder<VariableQuery>> fn) {
-    return search(SearchQueryBuilders.variableSearchQuery(fn));
+    return search(variableSearchQuery(fn));
   }
 
   public VariableEntity getByKey(final Long key) {
-    final SearchQueryResult<VariableEntity> result =
-        search(SearchQueryBuilders.variableSearchQuery().filter(f -> f.variableKeys(key)).build());
-    if (result.total() < 1) {
-      throw new NotFoundException(String.format("Variable with key %d not found", key));
-    } else if (result.total() > 1) {
-      throw new CamundaSearchException(
-          String.format("Found Variable with key %d more than once", key));
-    } else {
-      return result.items().stream().findFirst().orElseThrow();
+    final var result =
+        variableSearchClient
+            .withSecurityContext(securityContextProvider.provideSecurityContext(authentication))
+            .searchVariables(variableSearchQuery(q -> q.filter(f -> f.variableKeys(key))));
+    final var variableEntity = getSingleResultOrThrow(result, key, "Variable");
+    final var authorization = Authorization.of(a -> a.processDefinition().readInstance());
```

Comment:
❓Will there be more fine grained authorizations like read variables, too? Someone could read process instance but not the variables.

Reply:
according to https://docs.google.com/document/d/1ikxw9ARCv-MBwASnj7WhW5uHCndCRk2B35xi-cB2N0M/edit?tab=t.0#heading=h.m2zg2f7pb01i the same permission should be checked